LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/Ipmi.cs

```diff
@@ -133,33 +131,34 @@ private unsafe void Update(StringBuilder stringBuilder)
         _voltages.Clear();
         _controls.Clear();
 
-        if (_sdr.Count == 0 || stringBuilder != null)
+        if (_sdrs.Count == 0 || stringBuilder != null)
         {
             byte[] sdrInfo = RunIPMICommand(COMMAND_GET_SDR_REPOSITORY_INFO, NETWORK_FUNCTION_STORAGE, new byte[] { });
-            if (sdrInfo[0] == 0)
+            if (sdrInfo?[0] == 0)
             {
-                int recordCount = sdrInfo[3] * 256 + sdrInfo[2];
+                int recordCount = (sdrInfo[3] * 256) + sdrInfo[2];
 
                 byte recordLower = 0;
                 byte recordUpper = 0;
                 for (int i = 0; i < recordCount; ++i)
                 {
                     byte[] sdrRaw = RunIPMICommand(COMMAND_GET_SDR, NETWORK_FUNCTION_STORAGE, new byte[] { 0, 0, recordLower, recordUpper, 0, 0xff });
-                    if (sdrRaw[0] == 0 && sdrRaw.Length >= 3)
+                    if (sdrRaw?.Length >= 3 && sdrRaw[0] == 0)
                     {
                         recordLower = sdrRaw[1];
                         recordUpper = sdrRaw[2];
 
-                        Interop.Ipmi.Sdr sdr;
-
                         fixed (byte* pSdr = sdrRaw)
                         {
-                            sdr = (Interop.Ipmi.Sdr)Marshal.PtrToStructure((IntPtr)pSdr + 3, typeof(Interop.Ipmi.Sdr));
+                            object sdrObj = Marshal.PtrToStructure((IntPtr)pSdr + 3, typeof(Interop.Ipmi.Sdr));
```

Comment:
I am not sure you why would check the return of the Marshal. That should either be a valid pointer, or it should throw an exception.